src/haz3lcore/statics/TermBase.re

```diff
@@ -243,6 +254,104 @@ and Exp: {
     | Cast(t, Typ.t, Typ.t)
   and t = IdTagged.t(term);
 
+  let rec term_of_menhir_ast = (exp: Haz3lmenhir.AST.exp): Exp.term => {
+    switch (exp) {
+    | InvalidExp(s) => Invalid(s)
+    | Int(i) => Int(i)
+    | Float(f) => Float(f)
+    | String(s) => String(s)
+    | Bool(b) => Bool(b)
+    | Var(x) => Var(x)
+    | Constructor(x, ty) => Constructor(x, Typ.of_menhir_ast(ty))
+    | Deferral(pos) =>
+      switch (pos) {
+      | InAp => Deferral(InAp)
+      | OutsideAp => Deferral(OutsideAp)
+      }
+    | ListExp(l) => ListLit(List.map(of_menhir_ast, l))
+    | TupleExp(t) =>
+      if (List.length(t) == 1) {
+        Parens(of_menhir_ast(List.hd(t)));
+      } else {
+        Tuple(List.map(of_menhir_ast, t));
+      }
+    | Let(p, e1, e2) =>
+      Let(Pat.of_menhir_ast(p), of_menhir_ast(e1), of_menhir_ast(e2))
+    | FixF(p, e) => FixF(Pat.of_menhir_ast(p), of_menhir_ast(e), None)
+    | TypFun(t, e) => TypFun(TPat.of_menhir_ast(t), of_menhir_ast(e), None)
+    | Undefined => Undefined
+    | TyAlias(tp, ty, e) =>
+      TyAlias(
+        TPat.of_menhir_ast(tp),
+        Typ.of_menhir_ast(ty),
+        of_menhir_ast(e),
+      )
+    | BuiltinFun(s) => BuiltinFun(s)
+    | DeferredAp(f, a) => DeferredAp(of_menhir_ast(f), [of_menhir_ast(a)])
+    | Fun(p, e, name_opt) =>
+      switch (name_opt) {
+      | Some(name_str) =>
+        Fun(
+          Pat.of_menhir_ast(p),
+          of_menhir_ast(e),
+          None,
+          Some(name_str ++ "+"),
+        )
+      | None => Fun(Pat.of_menhir_ast(p), of_menhir_ast(e), None, None)
+      }
+    | ApExp(e1, e2) =>
+      Ap(Operators.Forward, of_menhir_ast(e1), of_menhir_ast(e2))
+    | BinExp(e1, op, e2) =>
+      BinOp(
+        Operators.of_menhir_ast(op),
+        of_menhir_ast(e1),
+        of_menhir_ast(e2),
+      )
+
+    | If(e1, e2, e3) =>
+      If(of_menhir_ast(e1), of_menhir_ast(e2), of_menhir_ast(e3))
+    | CaseExp(e, l) =>
+      let d_scrut = of_menhir_ast(e);
+      let d_rules =
+        List.map(
+          ((pat, exp)) => (Pat.of_menhir_ast(pat), of_menhir_ast(exp)),
+          l,
+        );
+      Match(d_scrut, d_rules);
+    | Cast(e, t1, t2) =>
+      Cast(of_menhir_ast(e), Typ.of_menhir_ast(t1), Typ.of_menhir_ast(t2))
+    | FailedCast(e, t1, t2) =>
+      FailedCast(
+        of_menhir_ast(e),
+        Typ.of_menhir_ast(t1),
+        Typ.of_menhir_ast(t2),
+      )
+    | EmptyHole => EmptyHole
+    | Seq(e1, e2) => Seq(of_menhir_ast(e1), of_menhir_ast(e2))
+    | Test(e) => Test(of_menhir_ast(e))
+    | Cons(e1, e2) => Cons(of_menhir_ast(e1), of_menhir_ast(e2))
+    | ListConcat(e1, e2) =>
+      ListConcat(of_menhir_ast(e1), of_menhir_ast(e2))
+    | Filter(a, cond, body) =>
+      let dcond = of_menhir_ast(cond);
+      let dbody = of_menhir_ast(body);
+      let act = FilterAction.of_menhir_ast(a);
+      Filter(StepperFilterKind.Filter({pat: dcond, act}), dbody);
+    | MultiHole(e) => MultiHole([Exp(of_menhir_ast(e))])
```

Comment:
Check to see if this should be a singleton list

Reply:
Sorry, what do you mean by this? I'm assuming MultiHole shouldn't just be a single Exp? If so, do you happen to know what a Multihole looks like (I'll also take a look at some in normal Hazel). Thanks!

Reply:
Yeah, that's what I mean. I'm not positive on what multiholes look like. I think they're caused by infix operators. I'll try to generate an example.

Reply:
This is the current visual representation of multiholes. It looks like your parser is surroudning the inhabitants with curly braces. We probably need some sort of delimiter to parse the children inside.
![Screenshot from 2024-08-26 14-21-55](https://github.com/user-attachments/assets/8c64a4b8-0175-40ee-96ce-6f406937d9c7)

Reply:
@green726 I talked to Cyrus about this some today and I think we can probably just skip the MultiHole case for now

Reply:
Ok, sounds good for me. I'll remove the multihole parsing for now.